src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -1709,6 +1709,48 @@ public class MyComponent : ComponentBase
         CompileToAssembly(generated);
     }
 
+    [IntegrationTestFact]
+    public void BindToComponent_SpecifiesValue_WithMatchingProperties_WithNameof()
```

Comment:
Do we have tests for situations like this

```razor
@using TheirComponent=Test.MyComponent;

<TheirComponent @bind-Value=""ParentValue"" />
@code {
    public int ParentValue { get; set; } = 42;

    public string nameof(string s) => string.Empty;
}");

```

Reply:
Sadly, this isn't supported today anyway: https://github.com/dotnet/razor/issues/7670

Reply:
> Sadly, this isn't supported today anyway: https://github.com/dotnet/razor/issues/7670

Locally it compiled fine on my machine. What am I missing?

Reply:
It'll compile, but you should get a warning ` warning RZ10012: Found markup element with unexpected name 'TheirComponent'. If this is intended to be a component, add a @using directive for its namespace.`